analysis/codelists.py

```diff
@@ -0,0 +1,451 @@
+    column="CTV3ID"
+)
+
+# Common covariate(s)
+
+# Ethnicity
+opensafely_ethnicity_codes_6 = codelist_from_csv(
+    "codelists/opensafely-ethnicity.csv",
+    column="Code",
+    category_column="Grouping_6"
+)
+
+primis_covid19_vacc_update_ethnicity = codelist_from_csv(
+    "codelists/primis-covid19-vacc-uptake-eth2001.csv",
+    column="code",
+    category_column="grouping_6_id"
+)
+
+# Smoking
+smoking_clear = codelist_from_csv(
+    "codelists/opensafely-smoking-clear.csv",
+    column="CTV3Code",
+    category_column="Category"
+)
+
+smoking_unclear = codelist_from_csv(
+    "codelists/opensafely-smoking-unclear.csv",
+    column="CTV3Code",
+    category_column="Category"
+)
+
+ever_current_smoke = codelist_from_csv(
+    "codelists/bristol-smoke-and-eversmoke.csv",
+    column="code"
+)
+
+# BMI
+bmi_obesity_snomed_clinical = codelist_from_csv(
+    "codelists/user-elsie_horne-bmi_obesity_snomed.csv",
+    column="code"
+)
+
+bmi_obesity_icd10 = codelist_from_csv(
+    "codelists/user-elsie_horne-bmi_obesity_icd10.csv",
+    column="code"
+)
+
+bmi_primis = codelist_from_csv(
+    "codelists/primis-covid19-vacc-uptake-bmi.csv",
+    column="code"
+)
+
+# Total Cholesterol
+cholesterol_snomed = codelist_from_csv(
+    "codelists/opensafely-cholesterol-tests-numerical-value.csv",
+    column="code"
+)
+
+# HDL Cholesterol
+hdl_cholesterol_snomed = codelist_from_csv(
+    "codelists/bristol-hdl-cholesterol.csv",
+    column="code"
+)
+
+# Carer codes
+carer_primis = codelist_from_csv(
+    "codelists/primis-covid19-vacc-uptake-carer.csv",
+    column="code"
+)
+
+# No longer a carer codes
+notcarer_primis = codelist_from_csv(
+    "codelists/primis-covid19-vacc-uptake-notcarer.csv",
+    column="code"
+)
+
+# Wider Learning Disability
+learndis_primis = codelist_from_csv(
+    "codelists/primis-covid19-vacc-uptake-learndis.csv",
+    column="code"
+)
+
+# Employed by Care Home codes
+carehome_primis = codelist_from_csv(
+    "codelists/primis-covid19-vacc-uptake-carehome.csv",
+    column="code"
+)
+
+# Employed by nursing home codes
+nursehome_primis = codelist_from_csv(
+    "codelists/primis-covid19-vacc-uptake-nursehome.csv",
+    column="code"
+)
+
+# Employed by domiciliary care provider codes
+domcare_primis = codelist_from_csv(
+    "codelists/primis-covid19-vacc-uptake-domcare.csv",
+    column="code"
+)
+
+# Patients in long-stay nursing and residential care
+longres_primis = codelist_from_csv(
+    "codelists/primis-covid19-vacc-uptake-longres.csv",
+    column="code"
+)
+
+# High Risk from COVID-19 code
+shield_primis = codelist_from_csv(
+    "codelists/primis-covid19-vacc-uptake-shield.csv",
+    column="code"
+)
+
+# Lower Risk from COVID-19 codes
+nonshield_primis = codelist_from_csv(
+    "codelists/primis-covid19-vacc-uptake-nonshield.csv",
+    column="code"
+)
+
+# For JCVI groups--------------------------------------------------------------------------
```

Comment:
Comment formatting throughout the script could be made more consistent but looks good otherwise.

Reply:
Done